internal/operands/tekton-pipelines/tekton-pipelines.go

```diff
@@ -45,46 +42,49 @@ type tektonPipelines struct {
 
 var _ operands.Operand = &tektonPipelines{}
 
-func New(bundle *tektonbundle.Bundle) *tektonPipelines {
-	tp := &tektonPipelines{
+func New(bundle *tektonbundle.Bundle) operands.Operand {
+	return &tektonPipelines{
 		pipelines:       bundle.Pipelines,
 		configMaps:      bundle.ConfigMaps,
 		roleBindings:    bundle.RoleBindings,
 		serviceAccounts: bundle.ServiceAccounts,
 		clusterRoles:    bundle.ClusterRoles,
 	}
-	return tp
 }
 
 func (t *tektonPipelines) Name() string {
 	return operandName
 }
 
-func (t *tektonPipelines) WatchClusterTypes() []client.Object {
-	return []client.Object{
-		&pipeline.Pipeline{},
-		&v1.ConfigMap{},
-		&rbac.RoleBinding{},
-		&v1.ServiceAccount{},
-		&rbac.ClusterRole{},
+func (t *tektonPipelines) WatchClusterTypes() []operands.WatchType {
+	return []operands.WatchType{
+		{Object: &pipeline.Pipeline{}, Crd: tektonCrd, WatchFullObject: true},
```

Comment:
This makes the teketon CRD required. SSP operator will wait until it exists.

@ksimon1, is it acceptable?

**future PR** - SSP should not wait for tekton CRD, if the feature gate is `false`.

Reply:
yes, that can be done follow up PR

Reply:
Hi, so if Tekton CRD doesn't exist, Tekton operands won't reconcile anymore.